stackslib/src/net/api/postblock_proposal.rs

```diff
@@ -0,0 +1,549 @@
+                );
+                return Err(BlockValidateReject {
+                    reason,
+                    reason_code: ValidateRejectCode::BadTransaction,
+                });
+            }
+        }
+
+        let mut block = builder.mine_nakamoto_block(&mut tenure_tx);
+        let size = builder.get_bytes_so_far();
+        let cost = builder.tenure_finish(tenure_tx);
+
+        // Clone signatures from block proposal
+        // These have already been validated by `validate_nakamoto_block_burnchain()``
+        block.header.miner_signature = self.block.header.miner_signature.clone();
+        block.header.signer_signature = self.block.header.signer_signature.clone();
+
+        // Assuming `tx_nerkle_root` has been checked we don't need to hash the whole block
+        let expected_block_header_hash = self.block.header.block_hash();
+        let computed_block_header_hash = block.header.block_hash();
+
+        if computed_block_header_hash != expected_block_header_hash {
+            warn!(
+                "Rejected block proposal";
+                "reason" => "Block hash is not as expected",
+                "expected_block_header_hash" => %expected_block_header_hash,
+                "computed_block_header_hash" => %computed_block_header_hash,
+                //"expected_block" => %serde_json::to_string(&serde_json::to_value(&self.block).unwrap()).unwrap(),
+                //"computed_block" => %serde_json::to_string(&serde_json::to_value(&block).unwrap()).unwrap(),
+            );
+            return Err(BlockValidateReject {
+                reason: "Block hash is not as expected".into(),
+                reason_code: ValidateRejectCode::BadBlockHash,
+            });
+        }
+
+        info!(
+            "Participant: validated anchored block";
+            "block_header_hash" => %computed_block_header_hash,
+            "height" => block.header.chain_length,
+            "tx_count" => block.txs.len(),
+            "parent_stacks_block_id" => %block.header.parent_block_id,
+            "block_size" => size,
+            "execution_cost" => %cost,
+            "validation_time_ms" => time_elapsed(),
+            "tx_fees_microstacks" => block.txs.iter().fold(0, |agg: u64, tx| {
+                agg.saturating_add(tx.get_tx_fee())
+            })
+        );
+
+        Ok(BlockValidateOk { block, cost, size })
+    }
+}
+
+#[derive(Clone, Default)]
+pub struct RPCBlockProposalRequestHandler {
+    pub block_proposal: Option<NakamotoBlockProposal>,
+}
+
+impl RPCBlockProposalRequestHandler {
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Decode a bare transaction from the body
+    fn parse_octets(mut body: &[u8]) -> Result<NakamotoBlockProposal, Error> {
+        NakamotoBlockProposal::consensus_deserialize(&mut body).map_err(|e| match e {
+            CodecError::DeserializeError(msg) => {
+                Error::DecodeError(format!("Failed to deserialize posted transaction: {msg}"))
```

Comment:
"transaction"?

Reply:
I'm removing `parse_octets` entirely, but yes, I'll clean up the "transaction" copypasta while working on the docs updates.

Reply:
Removed in aad22e9